subsys/bluetooth/host/conn.c

```diff
@@ -1803,19 +1803,24 @@ struct bt_conn *bt_conn_lookup_handle(u16_t handle)
 	return NULL;
 }
 
-int bt_conn_addr_le_cmp(const struct bt_conn *conn, const bt_addr_le_t *peer)
+bool bt_conn_is_peer_addr_le(const struct bt_conn *conn, u8_t id,
+			     const bt_addr_le_t *peer)
 {
+	if (id != conn->id) {
+		return false;
+	}
+
 	/* Check against conn dst address as it may be the identity address */
 	if (!bt_addr_le_cmp(peer, &conn->le.dst)) {
-		return 0;
+		return true;
 	}
 
 	/* Check against initial connection address */
 	if (conn->role == BT_HCI_ROLE_MASTER) {
 		return bt_addr_le_cmp(peer, &conn->le.resp_addr);
```

Comment:
Don't you need to change this line as well? I'd prefer if you do explicit comparisons with the return value as well, so `return (bt_addr_le_cmp(...) == 0);` here as well as a few lines below

Reply:
Yes missed that, thank you.

Reply:
And just to clarify why I prefer the explicit comparisons: normally what you'd apply the `!` operator to is a boolean expression, however in this case you're applying it to a comparison function that can return negative, 0, or positive, so it makes the intent of the code harder to follow.

Reply:
Just to clarify, do mean the return statements, or the if statement as well, because the if's all use `if (!memcmp())` and is fairly consistent.

Reply:
Yeah, I know. If I were to write everything from scratch I might prefer to have the `== 0` everywhere. At leasts for if-statments its immediately clear what's expected (it's always a boolean expression) and I'm also quite used to seeing the `if (!memcmp())` style stuff there so that it's immediately clear. In the case of the return statement you'd have to jump up to the beginning of the function to realise that it returns a boolean type, so being more explicit is more justified there, IMO. Feel free to disagree 😉